Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -132,6 +132,26 @@ lemma IsIntegralCurveAt.isIntegralCurveOn (h : ∀ t ∈ s, IsIntegralCurveAt γ
   obtain ⟨s, hs, h⟩ := isIntegralCurveAt_iff.mp (h t ht)
   exact h t (mem_of_mem_nhds hs)
 
+lemma IsIntegralCurveOn.congr_of_eqOn (hs : IsOpen s) (h : IsIntegralCurveOn γ v s)
+    (hγ : EqOn γ γ' s) : IsIntegralCurveOn γ' v s := by
+  intros t ht
+  rw [← hγ ht]
+  apply (h t ht).congr_of_eventuallyEq
+  exact Filter.eventuallyEq_of_mem (hs.mem_nhds ht) hγ.symm
+
```

Comment:
Let's also add:
```suggestion

lemma IsIntegralCurveOn.congr_of_eqOn_iff (hs : IsOpen s) (hγ : EqOn γ γ' s) :
    IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn γ' v s :=
  ⟨fun h ↦ IsIntegralCurveOn.congr_of_eqOn hs h hγ,
   fun h ↦ IsIntegralCurveOn.congr_of_eqOn hs h hγ.symm⟩

```

Reply:
I also added the iff version of `congr_of_eventuallyEq` and changed the spelling of the lemma name.